### ISSUE TITLE
[Bug]: why does netclient.service waits for 17 seconds?

### ISSUE THREAD

**Issue by mfide** [gravitl/netmaker#3019]
### Contact Details

mehmet.fide@gmail.com

### What happened?

I have netclient v0.24.2 runs on Ubuntu 24.04 server.

When I run `systemd-analyze blame` to see the slowest services in my system, I get following list:
`17.069s netclient.service
 7.630s systemd-networkd-wait-online.service
 2.251s fwupd-refresh.service
 2.109s fwupd.service
 1.101s lvm2-monitor.service
  854ms docker.service
  775ms snapd.seeded.service
  695ms snapd.service
..`

netclient.service is the top1 in this list. 
Then I check the content of `/etc/systemd/system/netclient.service` file:
`[Unit]
Description=Netclient Daemon
Documentation=https://docs.netmaker.org https://k8s.netmaker.org
After=network-online.target
Wants=network-online.target

[Service]
User=root
Type=simple
ExecStartPre=/bin/sleep 17
ExecStart=/sbin/netclient daemon
Restart=on-failure
RestartSec=15s

[Install]
WantedBy=multi-user.target`

There is a PreStart script which waits for **17 seconds** as hard coded for nothing in the netclient.service file.
According to the blame list above, my system becomes online in 7 seconds (systemd-networkd-wait-online service).

Why does netclient service has this 17 second sleep time?

Thank you.

### Version

v0.24.2

### What OS are you using?

Linux

### Contributing guidelines

- [X] Yes, I did.

**Comment by Cirr0e** [comment]
Hi there! I can help explain why this delay exists and what we can do about it.

Looking at your service file, that 17-second sleep delay is actually a legacy workaround from earlier versions of netclient. It was originally implemented to ensure network services were fully initialized before starting netclient, but this approach has been largely superseded by better handling in newer versions.

Since you're running v0.24.2 and your system shows network-online is achieved in ~7.6 seconds, this delay is unnecessarily long. The service file already has the proper dependencies:

```
After=network-online.target
Wants=network-online.target
```

You can safely modify this by:

1. Creating an override file:
```bash
sudo systemctl edit netclient.service
```

2. Add these lines to reduce or remove the delay:
```
[Service]
ExecStartPre=
ExecStartPre=/bin/sleep 5
```

This will override the original 17-second sleep with a more reasonable 5-second delay. I'm suggesting 5 seconds rather than removing it entirely just to provide a small buffer after network-online.target is reached.

References:
- Issue #1524 confirmed this behavior on similar systemd distributions
- Issue #744 showed that newer versions (post 0.14.0) handle network initialization much more gracefully
- Issue #53 tracked the historical context of these timing issues

Important considerations:
- If you experience any network connectivity issues after reducing the delay, you can adjust the sleep time accordingly
- Make sure to test the changes during a maintenance window
- Use `systemctl restart netclient` to apply the changes

Let me know if you need any clarification or run into any issues!
